src/v1/unpack.rs

```diff
@@ -244,43 +244,30 @@ impl<'a> PackStreamDecoder<'a> {
         Ok(byte)
     }
 
-    fn read_2_bytes(&mut self) -> PyResult<[u8; 2]> {
-        Ok([self.read_byte()?, self.read_byte()?])
-    }
-
-    fn read_4_bytes(&mut self) -> PyResult<[u8; 4]> {
-        Ok([
-            self.read_byte()?,
-            self.read_byte()?,
-            self.read_byte()?,
-            self.read_byte()?,
-        ])
-    }
-
-    fn read_8_bytes(&mut self) -> PyResult<[u8; 8]> {
-        Ok([
-            self.read_byte()?,
-            self.read_byte()?,
-            self.read_byte()?,
-            self.read_byte()?,
-            self.read_byte()?,
-            self.read_byte()?,
-            self.read_byte()?,
-            self.read_byte()?,
-        ])
+    fn read_n_bytes<const N: usize>(&mut self, n: usize) -> PyResult<[u8; N]> {
+        let from = self.index;
+        self.index += n;
+        unsafe {
```

Comment:
For every unsafe block, I prefer to have a comment stating what safety invariant needs to be upheld and how/why we do so.

That makes it easier to verify and to keep the code safe in the future.